Orange/projection/freeviz.py

```diff
@@ -0,0 +1,467 @@
+
+import numpy as np
+import scipy.spatial
+
+from Orange.data import ContinuousVariable, Domain
+from Orange.data.util import SharedComputeValue
+from Orange.projection import Projector, Projection
+
+
+class FreeVizProjector(SharedComputeValue):
+    """Transform into a given FreeViz component."""
+
+    def __init__(self, projection, feature, freeviz_transform):
+        super().__init__(freeviz_transform)
+        self.projection = projection
+        self.feature = feature
+        self.transformed = None
+
+    def compute(self, data, freeviz_space):
+        if self.feature >= freeviz_space.shape[1]:
+            return np.zeros((freeviz_space.shape[0]))
+        return freeviz_space[:, self.feature]
+
+
+class _FreeVizTransformDomain:
+    """Computation common for all FreeViz variables."""
+
+    def __init__(self, freeviz):
+        self.freeviz = freeviz
+
+    def __call__(self, data):
+        if data.domain != self.freeviz.pre_domain:
+            data = data.transform(self.freeviz.pre_domain)
+        return self.freeviz.transform(data.X)
+
+
+class _LinearCombination:
```

Comment:
@lanzagar, could `Orange.projection.pca._LinearCombination` be moved to Orange.projection.base.LinearCombination`, so it could be reused here?

Also, `PCAModel` is just a specific case of linear model. Could the common functionality be extracted into a base class? Could you and @jerneju discuss this? I don't insist it should be done, I'm just suggesting considering it. And it doesn't have to be in this PR, either.

Reply:
1) Could and should.
2) Could. Probably should? Definitely should be considered.

We'll take a look.